src/cli/cli.hpp

```diff
@@ -313,6 +314,93 @@ class Interpreter
         otError (Interpreter::*mHandler)(uint8_t aArgsLength, Arg aArgs[]);
     };
 
+    template <typename ValueType> using GetHandler         = ValueType (&)(otInstance *);
+    template <typename ValueType> using SetHandler         = void (&)(otInstance *, ValueType);
+    template <typename ValueType> using SetHandlerFailable = otError (&)(otInstance *, ValueType);
+
+    template <typename ValueType> otError ProcessGet(uint8_t aArgsLength, GetHandler<ValueType> aGetHandler)
+    {
+        static_assert(TypeTraits::IsSame<ValueType, uint8_t>::kValue ||
+                          TypeTraits::IsSame<ValueType, uint16_t>::kValue ||
+                          TypeTraits::IsSame<ValueType, uint32_t>::kValue,
+                      "ValueType must be uint8_t, uint16_t, or uint32_t");
```

Comment:
What about signed versions to support things like `otThreadGetParentPriority()` and `otThreadSetParentPriority()`?

Reply:
I added support for `int` value types as well but we don't have currently many cases where we can use them.